### PR TITLE
signalk-request and AP framework 

### PR DESCRIPTION
**Pull request by godind**
Hi @mxtommy ,

I have a few good changes to merge:

1. added Volume to units for tanks and such
2. improved signalk-request class and added filtering enabling features. If you had multiple put type Widget and subscribed to put Observer, all subscribers were getting all Subject. I added WidgetUUDI and RequestId enabling filter at widget, and/or on individual Puts. See commandReceived() in AP for a usage patern.
3. updated Switch and State Widgets accordingly
4. draft AP widget

I think we should merge so these fundamentals get merge sooner rather then later.

Cheers!

### CODE REVIEW

File: src/app/signalk-requests.service.ts [hunk 1]
```diff
@@ -50,48 +82,89 @@ export class SignalkRequestsService {
...
+     * @param path SignalK full path. Automatically removes "self" if included in path.
+     * @param value Value to be sent.
+     * @param widgetUUID Optional - Subscriber's Widget UUID to be included as part of
+     * the subscribeRequest Subject response. Enables Widget specific filtering.
+     * @return requestId Identifier for this specific request. Enables Request specific filtering.
+     */
+  public putRequest(path: string, value: any, widgetUUID: string): string {
     let requestId = this.newUuid();
-    let noSelfPath = path.replace(/^(self\.)/,""); //no self in path...
+    let selfContext: string = "vessels.self";    // TODO: hard coded context. Could be dynamic at some point
```
Comment (mxtommy) [inline src/app/signalk-requests.service.ts:99]: I'm pretty sure the spec doesn't use vessels.self anymore? Where did you see that?

**Comment by mxtommy** [comment]
Looks good! I added a new commit after the merge, mainly to put the delay in resize back into the steelGauges. Otherwise it re-initilizes the gauge as you resize, which is very laggy (it's not the most efficient gauge :D) Just be sure to pull the commit to your branch! :)

File: src/app/signalk-requests.service.ts (hunk 1 above)
Comment (godind) [reply src/app/signalk-requests.service.ts:99]: I connected with cmolete (he did the signalK-autopilot) and he help me out. It PUT request need a `vessels.self` context, but no `self.xxx` in the path. I tested and it work great.

**Comment by godind** [comment]
Great. I'll pull and merge master again.

File: src/app/signalk-requests.service.ts (hunk 1 above)
Comment (godind) [reply src/app/signalk-requests.service.ts:99]: Oh and I just noticed, you don;t have my latest version. Next pull: I'll first integrate Widget notification to your notification framework. Easier that way for you to see where I'm going with this...

**Pull request merged**